### PR TITLE
First pass at  benchmark Github action

### PR DESCRIPTION
**Pull request by atomaka**
See https://github.com/OAK-Foundation/OAK-blockchain/pull/254 for a sample PR created by this request.

### CODE REVIEW

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            --extrinsic '*' \
+            --repeat 20 \
+            --steps 50 \
+            --output ./${{ matrix.pallet }}-raw-weights.rs \
+            --template ./.maintain/frame-weight-template.hbs
+      - name: Upload raw weights
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.pallet }}
+          path: ${{ matrix.pallet }}-raw-weights.rs
```
Comment (nvengal) [inline .github/workflows/benchmark.yml:66]: Any reason to have retention policy here as well?

**Review by nvengal — APPROVED** [review]

File: .github/workflows/benchmark.yml (hunk 1 above)
Comment (atomaka) [reply .github/workflows/benchmark.yml:66]: The default of 90 days seemed reasonable.  The only reason the `oak-collator` has one is that I don't consider it an artifact of the job. I just abuse artifacts to pass a specific version of the binary to a later job for use.

**Review by rhuttman — COMMENTED** [review]
do you have an example output file for this?

**Comment by atomaka** [comment]
> do you have an example output file for this?

@rhuttman The PR linked in the description (https://github.com/OAK-Foundation/OAK-blockchain/pull/254) is *basically* the output. The only difference is the I reverted the struct name change to match your PR. For completeness, here's one I just generated locally using this branch: https://gist.github.com/atomaka/484cef3331baa00ccbb470abe14527ea

**Review by rhuttman — APPROVED** [review]

**Pull request merged**